### PR TITLE
CXP-1354: Display errors messages

### PR DESCRIPTION
**Pull request by eoliveauakeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/front/src/components/CatalogEdit/hooks/useCatalogForm.ts [hunk 1]
```diff
@@ -68,9 +68,16 @@ export const useCatalogForm = (id: string): Result => {
                         dispatch(action);
                         break;
                     case CatalogFormActions.SET_PRODUCT_SELECTION_CRITERIA:
-                        if (Object.keys(values.product_selection_criteria).length > Object.keys(action.value).length) {
```
Comment (eoliveauakeneo) [inline components/catalogs/front/src/components/CatalogEdit/hooks/useCatalogForm.ts:71]: @tseho after discussing it with @masacc I removed this control.

On "SET_PRODUCT_MAPPING" we can't have this kind of check and we will remove all the mapping errors on any changes.

The code is easier if the 2 cases have the same behavior, could we have your take on this ?

File: components/catalogs/front/src/components/ProductMapping/components/SelectAttributeDropdown.tsx [hunk 2]
```diff
@@ -97,6 +97,11 @@ export const SelectAttributeDropdown: FC<Props> = ({code, onChange, error}) => {
                         </Dropdown.Overlay>
                     )}
                 </Dropdown>
+                {undefined !== error && (
+                    <Helper inline level="error">
+                        {error}
```
Comment (masacc) [inline components/catalogs/front/src/components/ProductMapping/components/SelectAttributeDropdown.tsx:102]: There will never be a translation of the error message with that way of doing things. Is it ok?

**Review by masacc — APPROVED** [review]

File: components/catalogs/front/src/components/CatalogEdit/hooks/useCatalogForm.ts (hunk 1 above)
Comment (RiffVentura) [reply components/catalogs/front/src/components/CatalogEdit/hooks/useCatalogForm.ts:71]: I'm wondering if the error setting should be handled here at all:
Correct me if I'm wrong but If there are errors on both product selection and product mapping, dispatch will erase errors on the other tab

Comment (eoliveauakeneo) [reply components/catalogs/front/src/components/CatalogEdit/hooks/useCatalogForm.ts:71]: I added a filter to keep the errors corresponding to other tabs.

File: components/catalogs/front/src/components/ProductMapping/components/SelectAttributeDropdown.tsx (hunk 2 above)
Comment (eoliveauakeneo) [reply components/catalogs/front/src/components/ProductMapping/components/SelectAttributeDropdown.tsx:102]: The error translation is handled php side, some were missing for the mapping and I added them like we discussed on slack.

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**